
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

    <title>Usage &#8212; ThermOptCOBRA  documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Installation" href="installation.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="usage">
<h1>Usage<a class="headerlink" href="#usage" title="Permalink to this heading">¶</a></h1>
<section id="building-a-model">
<h2>Building a model<a class="headerlink" href="#building-a-model" title="Permalink to this heading">¶</a></h2>
<p><strong>CarveMe</strong> provides a very simple command line interface to build models.
The most basic utilization is:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa
</pre></div>
</div>
<p>This will build a genome-scale metabolic model from the <strong>genome</strong> file.</p>
<p>By default <strong>CarveMe</strong> expects a protein FASTA file. Alternatively, you can also provide DNA sequences instead:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>--dna<span class="w"> </span>genome.fna
</pre></div>
</div>
<p>Note that raw genome files are not supported. The FASTA file must be divided into individual genes.</p>
<p>It is possible to specify a different name or directory for the output file:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>--output<span class="w"> </span>model.xml
</pre></div>
</div>
<p>Short version:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>-o<span class="w"> </span>model.xml
</pre></div>
</div>
<p>If you want to produce a compressed SBML file, just change the extension (this is automatically supported by libSBML):</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>-o<span class="w"> </span>model.xml.gz
</pre></div>
</div>
<p>Rather than providing the genome data yourself, you can also provide an NCBI RefSeq accession code.
This will automatically download the sequence and build the model:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>--refseq<span class="w"> </span>GCF_000005845.2<span class="w"> </span>-o<span class="w"> </span>ecoli_k12_mg1655.xml
</pre></div>
</div>
<p>If you have downloaded multiple genome sequences, you can run <em>recursive mode</em> to build multiple models in one call.</p>
<p>This will launch multiple parallel processes, which can decrease the overall computation time if you are running
<strong>CarveMe</strong> in a multi-core CPU or in a computing cluster:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>-r<span class="w"> </span>myfolder/*.faa
</pre></div>
</div>
<p>This can be combined with <em>-o</em> to change the output folder:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>-r<span class="w"> </span>myfolder/*.faa<span class="w"> </span>-o<span class="w"> </span>mymodels/
</pre></div>
</div>
</section>
<section id="gap-filling">
<h2>Gap Filling<a class="headerlink" href="#gap-filling" title="Permalink to this heading">¶</a></h2>
<p><strong>CarveMe</strong> tries to predict the uptake and secretion capabilities of an organism only from genetic evidence,
and will produce a simulation-ready model without gap-filling for any particular media.</p>
<p>However, there are situations where you want to guarantee that the model is able to reproduce growth in one, or several,
experimentally verified media.</p>
<p>For instance, you can ensure the model reproduces growth on M9 and LB media:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>--gapfill<span class="w"> </span>M9,LB
</pre></div>
</div>
<p>Short version:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>-g<span class="w"> </span>M9,LB
</pre></div>
</div>
<p>Please see the <em>Advanced Usage</em> section on how to provide your own media compositions.</p>
<p>If you already have a model, and you just want to gap-fill it, you can do it with the <em>gapfill</em> utility function:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>gapfill<span class="w"> </span>model.xml<span class="w"> </span>-m<span class="w"> </span>M9<span class="w"> </span>-o<span class="w"> </span>new_model.xml
</pre></div>
</div>
<p>Please note that the result is not the same if you gap-fill during reconstruction. When you gap-fill during
reconstruction, the gene annotation scores are used to prioritize the reactions selected for gap-filling based on
genetic evidence. If you invoke <em>gapfill</em> alone, all potential gap-filling reactions are treated equally.</p>
<p>Finally, it is important to note that the models generated with <strong>CarveMe</strong> are not initialized with any
medium composition.</p>
<p>You can define the growth environment of the organism for simulation purposes by setting the flux bounds
of the exchange reactions yourself to match the respective medium composition.</p>
<p>Alternatively, you can tell <strong>CarveMe</strong> you want the model to come with a pre-defined medium composition.</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>--init<span class="w"> </span>M9
</pre></div>
</div>
<p>Short version:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>-i<span class="w"> </span>M9
</pre></div>
</div>
<p>Note that this will not gap-fill the model, but only define the external environment for simulation purposes.</p>
<p>To simultaneously gap-fill and initialize the model for a desired medium, you must combine both flags:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>carve<span class="w"> </span>genome.faa<span class="w"> </span>-g<span class="w"> </span>M9<span class="w"> </span>-i<span class="w"> </span>M9
</pre></div>
</div>
<p>You are now a basic user. Happy <em>carving</em>!</p>
</section>
<section id="microbial-communities">
<h2>Microbial Communities<a class="headerlink" href="#microbial-communities" title="Permalink to this heading">¶</a></h2>
<p><strong>CarveMe</strong> enables the generation of microbial community models from single species models.</p>
<p>The most basic usage is:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>merge_community<span class="w"> </span>organism_1.xml<span class="w"> </span>organism_2.xml<span class="w"> </span>...<span class="w"> </span>organism_N.xml<span class="w"> </span>-o<span class="w"> </span>community.xml
</pre></div>
</div>
<p>or more simply:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>merge_community<span class="w"> </span>*.xml<span class="w"> </span>-o<span class="w"> </span>community.xml
</pre></div>
</div>
<p>This generates an SBML file with a community where each organism is assigned to its own compartment and
a common community biomass equation is also generated. You can import the merged model into any simulation tool, just
as any normal constraint-based model and apply different types of simulation methods (FBA, FVA, etc…).
You can initialize the community with a pre-defined medium (just like during single-species reconstruction):</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>merge_community<span class="w"> </span><span class="o">[</span>input<span class="w"> </span>files<span class="o">]</span><span class="w"> </span>-i<span class="w"> </span>M9
</pre></div>
</div>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">ThermOptCOBRA</a></h1>








<h3>Navigation</h3>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Usage</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#building-a-model">Building a model</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gap-filling">Gap Filling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#microbial-communities">Microbial Communities</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="installation.html" title="previous chapter">Installation</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 6.2.1</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 0.7.16</a>
      
      |
      <a href="_sources/usage.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>